tests/test_wires.py

```diff
@@ -0,0 +1,253 @@
+        assert wires.wire_list == [wire]
+
+    @pytest.mark.parametrize("iterable", [np.array([4, 1, 1, 3]),
+                                          [4, 1, 1, 3],
+                                          (4, 1, 1, 3),
+                                          ['a', 'a', 'b'],
+                                          [Wires([1, 0]), Wires([1, 0]), Wires([3])]])
+    def test_error_for_repeated_wires(self, iterable):
+        """Tests that a Wires object cannot be created from iterables with repeated indices."""
+
+        with pytest.raises(WireError, match="Wires must be unique"):
+            Wires(iterable)
+
+    @pytest.mark.parametrize("iterable", [[4, 1, 0, 3],
+                                          ['a', 'b', 'c']])
+    def test_indexing_and_slicing(self, iterable):
+        """Tests the indexing and slicing of Wires objects."""
+
+        wires = Wires(iterable)
+
+        # check single index
+        for i in range(len(iterable)):
+            assert wires[i] == Wires(iterable[i])
+        # check slicing
+        assert wires[:2] == Wires(iterable[:2])
+
+    def test_equality(self):
+        """Tests that we can compare Wires objects with the '==' and '!=' operators."""
+
+        wires1 = Wires([1, 2, 3])
+        wires2 = Wires([3, 2, 1])
+        wires3 = Wires([1, 2, 3])
+        assert wires1 != wires2
+        assert wires1 == wires3
+
+    def test_is_ordered(self):
+        """Tests that a Wires object is not equal to another Wires object with a different ordering of the indices."""
+
+        wires1 = Wires([1, 2, 3])
+        wires2 = Wires([3, 2, 1])
+        assert wires1 != wires2
+
+    def test_length(self):
+        """Tests that a Wires object returns the correct length."""
+
+        wires = Wires([1, 2, 3, 4, 5])
+        assert len(wires) == 5
+
+    def test_representation(self):
+        """Tests the string representation."""
+
+        wires_str = str(Wires([1, 2, 3]))
+
+        assert wires_str == "<Wires = {}>".format([1, 2, 3])
+
+    def test_set(self):
+        """Tests that the implementation of __hash__ allows for the set() function to work."""
+
+        wires = Wires([0, 1, 2])
+        list_of_wires = [Wires([1]), Wires([1]), Wires([1, 2, 3]), Wires([4])]
+
+        assert set(wires) == {Wires([0]), Wires([1]), Wires([2])}
+        assert set(list_of_wires) == {Wires([1]), Wires([1, 2, 3]), Wires([4])}
+
+    def test_convert_to_numpy_array(self):
+        """Tests that Wires object can be converted to a numpy array."""
+
+        wires = Wires([4, 0, 1])
+        array = wires.as_ndarray()
+        assert isinstance(array, np.ndarray)
+        assert array.shape == (3, )
+        for w1, w2 in zip(array, np.array([4, 0, 1])):
+            assert w1 == w2
+
+    def test_convert_to_list(self):
+        """Tests that Wires object can be converted to a list."""
+
+        wires = Wires([4, 0, 1])
+        lst = wires.as_list()
+        assert isinstance(lst, list)
```

Comment:
Not at all important here, but just as a python protip, it's also common to use e.g., `list_` for the name of something that shadows a built-in name